platforms/nuttx/NuttX/CMakeLists.txt

```diff
@@ -180,10 +179,48 @@ function(add_nuttx_dir nuttx_lib nuttx_lib_dir kernel extra target)
 	add_dependencies(nuttx_${nuttx_lib} nuttx_${nuttx_lib}_build)
 endfunction()
 
+# helper for drivers with segger_sysview patches
+function(add_nuttx_dir_patched nuttx_lib nuttx_lib_dir kernel extra target)
+	if (${target} STREQUAL all)
+		set(nuttx_lib_target all)
+	else()
+		set(nuttx_lib_target lib${target}.a)
+	endif()
+
+	file(GLOB_RECURSE nuttx_lib_files LIST_DIRECTORIES false
+		${CMAKE_CURRENT_SOURCE_DIR}/nuttx/${nuttx_lib_dir}/*.c
+		${CMAKE_CURRENT_SOURCE_DIR}/nuttx/${nuttx_lib_dir}/*.h
+	)
+
+	add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/nuttx/${nuttx_lib_dir}/lib${nuttx_lib}.a
+		COMMAND ${CMAKE_COMMAND} -E remove -f ${NUTTX_DIR}/${nuttx_lib_dir}/lib${nuttx_lib}.a
+		COMMAND find ${nuttx_lib_dir} -type f -name \*.o -delete
+		COMMAND make -C ${nuttx_lib_dir} --no-print-directory --silent ${nuttx_lib_target} TOPDIR="${NUTTX_DIR}" KERNEL=${kernel} EXTRAFLAGS=${extra}
+		COMMAND ${CMAKE_OBJCOPY} ${NUTTX_DIR}/${nuttx_lib_dir}/lib${nuttx_lib}.a --redefine-sym sched_note_start=segger_note_start --redefine-sym sched_note_stop=segger_note_stop --redefine-sym sched_note_suspend=segger_note_suspend --redefine-sym sched_note_resume=segger_note_resume ${NUTTX_DIR}/${nuttx_lib_dir}/lib${nuttx_lib}.a
```

Comment:
Can we simply have different scheduling notes built based on CONFIG_SEGGER_SYSVIEW or is there a reason it needs to be patched like this?

Reply:
So both cpuload and sysview use the NuttX note hooks, therefore functions like `sched_note_start` are duplicate. The trick what I do here is too rename the `sched_note_start` from sysview into `segger_note_start` and call function from the cpuload `sched_note_start` to make sure that both PX4 cpuload works and sysview.

Reply:
Makes sense, I was just wondering if we could do it without duplicating this big ugly wad of cmake.

Reply:
Well @davids5 suggested another solution to use the linker wrap functionality to wrap the `sched_note_xxx` instances during linking process and then we just call the wrapped function from cpuload.cpp. I've tried this but linking in the PX4 CMake is rather complex.